src/tools/compiletest/src/header.rs

```diff
@@ -73,19 +73,24 @@ impl EarlyProps {
                 return false;
             }
 
-            if parse_name_directive(line, "ignore-gdb") {
+            if !line.contains("ignore-gdb-version") &&
+               parse_name_directive(line, "ignore-gdb") {
                 return true;
             }
 
             if let Some(actual_version) = config.gdb_version {
                 if line.contains("min-gdb-version") {
-                    let min_version = line.trim()
-                        .split(' ')
-                        .last()
-                        .expect("Malformed GDB version directive");
+                    let min_version = extract_gdb_version_range(line);
```

Comment:
I think the name is misleading because it is both the min and max version? Rather, than renaming, using a destructuring `let` rather than tuple indexing would probably be more readable.